avaframe/com1DFAPy/com1DFA.py

```diff
@@ -151,55 +151,81 @@ def initializeSimulation(cfg, relRaster, dem):
     partPerCell = np.zeros(np.shape(relRaster), dtype=np.int64)
     FD = np.zeros((nrows, ncols))
-    Npart = 0
-    NPPC = np.empty(0)
-    Apart = np.empty(0)
-    Xpart = np.empty(0)
-    Ypart = np.empty(0)
-    Mpart = np.empty(0)
-    Hpart = np.empty(0)
-    InCell = np.empty((0), int)
-    IndX = np.empty((0), int)
-    IndY = np.empty((0), int)
     # find all non empty cells (meaning release area)
     indRelY, indRelX = np.nonzero(relRaster)
-    # loop on non empty cells
-    for indRelx, indRely in zip(indRelX, indRelY):
-        # compute number of particles for this cell
-        h = relRaster[indRely, indRelx]
-        Vol = A[indRely, indRelx] * h
-        mass = Vol * rho
-        xpart, ypart, mPart, nPart = placeParticles(mass, indRelx, indRely, csz, massPerPart)
-        Npart = Npart + nPart
-        partPerCell[indRely, indRelx] = nPart
-        # initialize field Flow depth
-        FD[indRely, indRelx] = h
-        # initialize particles position, mass, height...
-        NPPC = np.append(NPPC, nPart*np.ones(nPart))
-        Apart = np.append(Apart, A[indRely, indRelx]*np.ones(nPart)/nPart)
-        Xpart = np.append(Xpart, xpart)
-        Ypart = np.append(Ypart, ypart)
-        Mpart = np.append(Mpart, mPart * np.ones(nPart))
-        Hpart = np.append(Hpart, h * np.ones(nPart))
-        ic = indRelx + ncols * indRely
-        IndX = np.append(IndX, np.ones(nPart)*indRelx)
-        IndY = np.append(IndY, np.ones(nPart)*indRely)
-        InCell = np.append(InCell, np.ones(nPart)*ic)
-
-    Hpart, _ = geoTrans.projectOnGrid(Xpart, Ypart, relRaster, csz=csz, interp='bilinear')
-    Mpart = rho * Hpart * Apart
-    # create dictionnary to store particles properties
-    particles = {}
-    particles['Npart'] = Npart
-    particles['NPPC'] = NPPC
-    particles['x'] = Xpart
-    particles['y'] = Ypart
-    particles['s'] = np.zeros(np.shape(Xpart))
-    # adding z component
-    particles, _ = geoTrans.projectOnRaster(dem, particles, interp='bilinear')
-    # readjust mass
-    mTot = np.sum(Mpart)
-    particles['m'] = Mpart*Mraster/mTot
+
+    # make option available to read initial particle distribution from file
+    if cfg.getboolean('flagInitializeFF'):
+        log.info('Initial particle distribution read from file!!')
+        inDirPart = os.path.join(avaDir, 'Inputs', 'particles')
+        Particles, TimeStepInfo = readPartFromPickle(inDirPart)
+        particles = Particles[0]
+        Xpart = particles['x']
+        Mpart = particles['m']
+        Hpart = np.ones(len(Xpart))
+        particles['Npart'] = len(Xpart)
+        particles = DFAfunC.getNeighboursC(particles, dem)
+        particles['s'] = np.zeros(np.shape(Xpart))
+        # # adding z component
+        zSamos = copy.deepcopy(particles['z'])
+        particles, _ = geoTrans.projectOnRasterVect(dem, particles, interp='bilinear')
+        plt.plot(particles['y'], zSamos, 'bo')
+        plt.plot(particles['y'], particles['z'], 'k+')
+        plt.show()
```

Comment:
I think this warants a new function initParticlesFromFile, right? Since this is something we might want to keep to experiment with....

Reply:
Can we address this issue in a next PR once we found a nicer way to get the particles initial position?